lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -0,0 +1,231 @@
+
+        add_invoice(post, money, options)
+      end
+
+      def add_encryption(post)
+        post[:cifrado] = CECA_ENCRIPTION
+      end
+
+      def add_signature(post, params_encoded, options)
+        post[:firma] = Digest::SHA2.hexdigest(@options[:signature_key].to_s + params_encoded)
+      end
+
+      def add_merchant_data(post)
+        params = post[:parametros] ||= {}
+
+        params[:merchantID] = @options[:merchant_id]
+        params[:acquirerBIN] = @options[:acquirer_bin]
+        params[:terminalID] = @options[:terminal_id]
+      end
+
+      def add_invoice(post, money, options)
+        post[:parametros][:numOperacion] = options[:operation_number] || options[:order_id]
+        post[:parametros][:importe] = amount(money)
+        post[:parametros][:tipoMoneda] = CECA_CURRENCIES_DICTIONARY[options[:currency] || currency(money)].to_s
+        post[:parametros][:exponente] = 2.to_s
+      end
+
+      def add_creditcard(post, creditcard)
+        params = post[:parametros] ||= {}
+
+        params[:pan] = creditcard.number
+        params[:caducidad] = strftime_yyyymm(creditcard)
+        params[:cvv2] = creditcard.verification_value
+        params[:csc] = creditcard.verification_value if CreditCard.brand?(creditcard.number) == 'american_express'
+      end
+
+      def add_stored_credentials(post, creditcard, options)
+        return unless stored_credential = options[:stored_credential]
+
+        return if options[:exemption_sca] == 'NONE' || options[:exemption_sca].blank?
+
+        params = post[:parametros] ||= {}
+        params[:exencionSCA] = options[:exemption_sca]
+
+        if options[:exemption_sca] == 'MIT'
+          requires!(stored_credential, :reason_type, :initiator)
+          requires!(options, :recurring_frequency)
+        end
+
+        params[:tipoCOF] = CECA_REASON_TYPES[stored_credential[:reason_type].to_sym]
+        params[:inicioRec] = CECA_INITIATOR[stored_credential[:initiator].to_sym]
+        params[:finRec] = options[:recurring_end_date] || strftime_yyyymm(creditcard)
+        params[:frecRec] = options[:recurring_frequency]
+
+        params[:mmppTxId] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
+      end
+
+      def add_three_d_secure(post, options)
+        params = post[:parametros] ||= {}
+
+        params[:ThreeDsResponse] = options[:three_d_secure] if options[:three_d_secure]
+      end
+
+      def commit(action, post, method = :post)
+        auth_options = {
+          operation_number: post[:parametros][:numOperacion],
+          amount: post[:parametros][:importe]
+        }
+
+        add_encryption(post)
+        add_merchant_data(post)
+
+        params_encoded = encode_params(post)
+        add_signature(post, params_encoded, options)
+
+        response = parse(ssl_request(method, url(action), post.to_json, headers))
+        response[:parametros] = parse(response[:parametros]) if response[:parametros]
```

Comment:
❓ Question:

So `parameters` is a JSON inside of the main JSON response?

Reply:
Yes it is

Reply:
Yes, it is